server/src/internalClusterTest/java/org/elasticsearch/snapshots/SnapshotStressTestsIT.java

```diff
@@ -0,0 +1,1578 @@
+                            }));
+                    }));
+                }))));
+
+                startedRestore = true;
+            } finally {
+                if (startedRestore == false) {
+                    startRestorer();
+                }
+            }
+        }
+
+        private void startCloner() {
+            enqueueAction(() -> {
+                boolean startedClone = false;
+                try (TransferableReleasables localReleasables = new TransferableReleasables()) {
+
+                    final List<TrackedSnapshot> trackedSnapshots = new ArrayList<>(snapshots.values());
+                    if (trackedSnapshots.isEmpty()) {
+                        return;
+                    }
+
+                    if (localReleasables.add(blockFullClusterRestart()) == null) {
+                        return;
+                    }
+
+                    final Client client = localReleasables.add(acquireClient()).getClient();
+
+                    final TrackedSnapshot trackedSnapshot = randomFrom(trackedSnapshots);
+                    if (localReleasables.add(trackedSnapshot.tryAcquirePermit()) == null) {
+                        return;
+                    }
+
+                    if (snapshots.get(trackedSnapshot.snapshotName) != trackedSnapshot) {
+                        // concurrently removed
+                        return;
+                    }
+
+                    final Releasable releaseAll = localReleasables.transfer();
+
+                    final StepListener<List<String>> getIndicesStep = new StepListener<>();
+
+                    logger.info(
+                        "--> listing indices in [{}:{}] in preparation for cloning",
+                        trackedSnapshot.trackedRepository.repositoryName,
+                        trackedSnapshot.snapshotName
+                    );
+
+                    trackedSnapshot.getSnapshotInfo(client, mustSucceed(snapshotInfo -> {
+                        final Set<String> failedShardIndices = snapshotInfo.shardFailures()
+                            .stream()
+                            .map(ShardOperationFailedException::index)
+                            .collect(Collectors.toSet());
+                        final Set<String> cloneableIndices = new HashSet<>(snapshotInfo.indices());
+                        cloneableIndices.removeAll(failedShardIndices);
+
+                        if (cloneableIndices.isEmpty()) {
+                            getIndicesStep.onResponse(Collections.emptyList());
+                            return;
+                        }
+
+                        if (failedShardIndices.isEmpty() && randomBoolean()) {
+                            getIndicesStep.onResponse(Collections.singletonList("*"));
+                            return;
+                        }
+
+                        final List<String> indexNames = new ArrayList<>();
+                        while (indexNames.isEmpty()) {
```

Comment:
Looks like we could just use randomFrom() here? Or even randomSubset()?

Reply:
:+1: good point, see 75acbf2.